### PR TITLE
DEVX-1531: add client config files and get all settings from configs across clients, update cloud client readmes

### PR DESCRIPTION
**Pull request by laura-brouckman**
Updated after discussion with @ybyzek and @rspurgeon 

- References configs that can be found here https://github.com/confluentinc/configuration-templates and will be the central source for the examples repo and ccloud ui 
- Updates readme's to reflect this change
- Future goal would be to add a script that can autogenerate these configs for them based on their cluster info

### CODE REVIEW

File: clients/cloud/c/README.md [hunk 1]
```diff
@@ -2,17 +2,25 @@
...
 
 * Access to a [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud) cluster
-* Local file with configuration parameters to connect to your Confluent Cloud instance ([how do I find those?](https://docs.confluent.io/current/cloud/using/config-client.html#librdkafka-based-c-clients?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud)).
+* Update the `example.config` file with the broker endpoint and api key to connect to your Confluent Cloud cluster ([how do I find those?](https://docs.confluent.io/current/cloud/using/config-client.html#librdkafka-based-c-clients?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud)).
```
Comment (ybyzek) [inline clients/cloud/c/README.md]: @laura-brouckman to find this information (broker end point, api key/secret), do you think we should point users to CCloud UI or to docs, or both?  We can make it a different PR, but I wanted to make sure we took into consideration updates that were made to the CCloud UI recently.

File: clients/cloud/c/example.config [hunk 2]
```diff
@@ -0,0 +1,5 @@
+bootstrap.servers=<BROKER_ENDPOINT>
```
Comment (ybyzek) [inline clients/cloud/c/example.config]: Some users thought these examples were for ccloud only, and so the README verbiage changed recently to make it more generic to clarify to users that these examples can be used for any Kafka cluster, not just CCloud.

To that end, should we create separate files, e.g.

* `example-ccloud.config`
* `exampe-localhost.config`

If we do consider this, we'll need to take into account whether CCloud expects a specific file name.

File: clients/cloud/clojure/README.md [hunk 3]
```diff
@@ -9,20 +9,19 @@ For more information, please see the [application development documentation](htt
...
-$ cat ~/.ccloud/example.config
-bootstrap.servers=<broker-1,broker-2,broker-3>
-sasl.username=<api-key-id>
-sasl.password=<secret-access-key>
+$ cat example.config
+bootstrap.servers=<BROKER_ENDPOINT>
 ssl.endpoint.identification.algorithm=https
 security.protocol=SASL_SSL
 sasl.mechanism=PLAIN
+sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username\="<CLUSTER_API_KEY>" password\="<CLUSTER_API_SECRET>";
```
Comment (ybyzek) [inline clients/cloud/clojure/README.md]: @laura-brouckman just confirming why the change...it is no longer `sasl.username` and `sasl.password`?

File: clients/cloud/groovy/README.adoc [hunk 4]
```diff
@@ -116,7 +116,7 @@ Verify that the consumer received all the messages:
 
  # Run the consumer
  $ ./gradlew runApp -PmainClass="io.confluent.examples.clients.cloud.StreamsExample" \
-   	-PconfigPath="$HOME/.ccloud/config" \
+   	-PconfigPath="$HOME/examples/clients/cloud/groovy/example.config" \
```
Comment (ybyzek) [inline clients/cloud/groovy/README.adoc]: 1. User may not put the `examples` repo in `$HOME/` so this may not always work as it is today.
2. Does it need the full path or would `-PconfigPath="./example.config"` work?

File: clients/cloud/java/README.md [hunk 5]
```diff
@@ -8,19 +8,21 @@ Produce messages to and consume messages from a Kafka cluster using the Java Pro
 * Java 1.8 or higher to run the demo application
 * Maven to compile the demo application
 
-To run this example, create a local file with configuration parameters to connect to your Kafka cluster, which can be on your local host, [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud), or any other cluster.
-If this is a Confluent Cloud cluster, you must have:
+To run this example, update the `example.config` file with configuration parameters to connect to your Kafka cluster, which can be on your local host, [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud), or any other cluster. If this is a Confluent Cloud cluster, you must have:
```
Comment (ybyzek) [inline clients/cloud/java/README.md]: It may be helpful to hyperlink all of example.config mentions.

For example, change:

```
`example.config`
```

to

```
[example.config](example.config)
```

File: clients/cloud/java/README.md [hunk 6]
```diff
@@ -8,19 +8,21 @@ Produce messages to and consume messages from a Kafka cluster using the Java Pro
...
 
 ```shell
-$ cat $HOME/.ccloud/config
-bootstrap.servers=<BROKER ENDPOINT>
+$ cat example.config
 ssl.endpoint.identification.algorithm=https
-security.protocol=SASL_SSL
 sasl.mechanism=PLAIN
-sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username\="<API KEY>" password\="<API SECRET>";
+request.timeout.ms=20000
```
Comment (ybyzek) [inline clients/cloud/java/README.md]: Remove all mentions of `request.timeout.ms` and `retry.backoff.ms` -- should not be needed any more per separate conversation.

File: clients/cloud/java/README.md [hunk 7]
```diff
@@ -107,7 +109,7 @@ The Kafka Streams API reads the same topic from Confluent Cloud and does a state
 
     # Run the Kafka streams application
     $ mvn exec:java -Dexec.mainClass="io.confluent.examples.clients.cloud.StreamsExample" \
-      -Dexec.args="$HOME/.ccloud/config test1"
+      -Dexec.args="$HOME/examples/clients/cloud/java/example.config test1"
```
Comment (ybyzek) [inline clients/cloud/java/README.md]: Same issue about full path as mentioned in the Groovy example.  Let's simplify to `example.config` if possible

File: clients/cloud/java/README.md [hunk 8]
```diff
@@ -149,24 +151,24 @@ Note that your VPC must be able to connect to the Confluent Cloud Schema Registr
...
+    $ curl -u <SR_API_KEY>:<SR_API_SECRET> https://<SR_ENDPOINT>/subjects
 
-    # Same as above, as a single bash command to parse the values out of $HOME/.ccloud/config
-    $ curl -u $(grep "^schema.registry.basic.auth.user.info" $HOME/.ccloud/config | cut -d'=' -f2) $(grep "^schema.registry.url" $HOME/.ccloud/config | cut -d'=' -f2)/subjects
+    # Same as above, as a single bash command to parse the values out of example.config
+    $ curl -u $(grep "^schema.registry.basic.auth.user.info" $HOME/examples/clients/cloud/java/example.config | cut -d'=' -f2) $(grep "^schema.registry.url" $HOME/examples/clients/cloud/java/example.config| cut -d'=' -f2)/subjects
     ```
 
-3. Add the following parameters to your local Confluent Cloud configuration file (``$HOME/.ccloud/config``). In the output below, substitute values for `<SR API KEY>`, `<SR API SECRET>`, and `<SR ENDPOINT>`.
+3. Add the following parameters to your local Confluent Cloud configuration file (``$HOME/examples/clients/cloud/java/example.config``). In the output below, substitute values for `<SR_API_KEY>`, `<SR_API_SECRET>`, and `<SR_ENDPOINT>`.
```
Comment (ybyzek) [inline clients/cloud/java/README.md]: For reasons stated earlier:

1. Simplify path to just example.config
2. Hyperlink

**Review by ybyzek — COMMENTED** [review]
@laura-brouckman thank you for creating this PR to improve the user experience running the client examples.

**Review by ybyzek — CHANGES_REQUESTED** [review]
@laura-brouckman for now marking "Request changes" to ensure we discuss offline the implications of this PR (see Slack convo).

File: clients/cloud/c/README.md (hunk 1 above)
Comment (laura-brouckman) [reply clients/cloud/c/README.md]: Yeah I was thinking about linking to the ccloud UI but the route has the cluster id in it (i.e. https://confluent.cloud/environments/t2333/clusters/lkc-5wqoq/settings/kafka) so I wasn't sure the best way to link there

File: clients/cloud/clojure/README.md (hunk 3 above)
Comment (laura-brouckman) [reply clients/cloud/clojure/README.md]: yeah - I was getting errors with sasl.username / password when running this example, and using the `sasl.jaas.config` instead fixed it

File: clients/cloud/c/README.md [hunk 9]
```diff
@@ -2,17 +2,26 @@
...
 Produce messages to and consume messages from a Kafka cluster using the C client [librdkafka](https://github.com/edenhill/librdkafka).
 
-
 # Prerequisites
 
 * [librdkafka](https://github.com/edenhill/librdkafka) installed on your machine, see [installation instructions](https://github.com/edenhill/librdkafka/blob/master/README.md#instructions).
 
-To run this example, create a local file with configuration parameters to connect to your Kafka cluster, which can be on your local host, [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud), or any other cluster.
-If this is a Confluent Cloud cluster, you must have:
+To run this example, download the `librdkafka.config` file from [confluentinc/configuration-templates](https://github.com/confluentinc/configuration-templates/tree/master/clients/cloud) and save it to a `$HOME/.ccloud` folder.
```
Comment (laura-brouckman) [inline clients/cloud/c/README.md]: Not sure if this is the best name for this config file

File: clients/cloud/clojure/README.md [hunk 10]
```diff
@@ -9,20 +9,19 @@ For more information, please see the [application development documentation](htt
 * Java 8 or higher (Clojure 1.10 recommends using Java 8 or Java 11)
 * The [Leiningen](https://leiningen.org/#install) tool to compile and run the demos
 
-To run this example, create a local file with configuration parameters to connect to your Kafka cluster, which can be on your local host, [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud), or any other cluster.
-If this is a Confluent Cloud cluster, you must have:
+To run this example, download the `java.config` file from [confluentinc/configuration-templates](https://github.com/confluentinc/configuration-templates/tree/master/clients/cloud) and save it to a `$HOME/.ccloud` folder.
```
Comment (laura-brouckman) [inline clients/cloud/clojure/README.md]: ditto on this config file name, this was to represent the `java-esque` ones but not sure what exactly to name it

File: clients/cloud/c/README.md (hunk 9 above)
Comment (ybyzek) [inline clients/cloud/c/README.md]: How about something like this:

```
* Create a local file (e.g. at `kafka-client.config`) with configuration parameters to connect to your Kafka cluster, which can be on your local host, [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud), or any other cluster.  Follow [these detailed instructions](https://github.com/confluentinc/configuration-templates/tree/master/clients/README.md) to properly create this local file.
```

What it accomplishes:

1. Provides usage in the README in https://github.com/confluentinc/configuration-templates/tree/master/clients/README.md (to be created...I can do this if you'd like)
2. Minimizes duplication in this repo
3. Maintains ability to support destination being CCloud or localhost or any other cluster

Then we could remove the lines below that start with "Update the configuration parameters" and "Update the " because that would go into the README

Comment (laura-brouckman) [reply clients/cloud/c/README.md]: 👍 makes sense, thanks for the suggestion

File: clients/cloud/confluent-cli/confluent-cli-example.sh [hunk 11]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-CONFIG_FILE=~/.ccloud/config
+CONFIG_FILE=~/.confluent/config
```
Comment (laura-brouckman) [inline clients/cloud/confluent-cli/confluent-cli-example.sh]: I changed `.ccloud` -> `.confluent` everywhere in the `.clients/cloud` subfolder, but can change these ones back if necessary

File: clients/cloud/c/README.md [hunk 12]
```diff
@@ -2,17 +2,25 @@
 
 Produce messages to and consume messages from a Kafka cluster using the C client [librdkafka](https://github.com/edenhill/librdkafka).
 
-
 # Prerequisites
 
 * [librdkafka](https://github.com/edenhill/librdkafka) installed on your machine, see [installation instructions](https://github.com/edenhill/librdkafka/blob/master/README.md#instructions).
+* Create a local file (e.g. at `librdkafka.config`) with configuration parameters to connect to your Kafka cluster, which can be on your local host, [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud), or any other cluster.  Follow [these detailed instructions](https://github.com/confluentinc/configuration-templates/tree/master/README.md) to properly create this file.
```
Comment (ybyzek) [inline clients/cloud/c/README.md]: @laura-brouckman , @rspurgeon 

did we want to go with `$HOME/.confluent/librdkafka.config` instead of `librdkafka.config`?

File: clients/cloud/c/README.md [hunk 13]
```diff
@@ -2,17 +2,25 @@
...
 
-To run this example, create a local file with configuration parameters to connect to your Kafka cluster, which can be on your local host, [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud), or any other cluster.
 If this is a Confluent Cloud cluster, you must have:
 
 * Access to a [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud) cluster
-* Local file with configuration parameters to connect to your Confluent Cloud instance ([how do I find those?](https://docs.confluent.io/current/cloud/using/config-client.html#librdkafka-based-c-clients?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud)).
 
+Additional configuration properties are supported, see [CONFIGURATION.md](https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md) for the full list.
```
Comment (ybyzek) [inline clients/cloud/c/README.md]: This line should be deleted now, right?

File: clients/cloud/c/README.md [hunk 14]
```diff
@@ -2,17 +2,25 @@
...
+
+```bash
```
Comment (ybyzek) [inline clients/cloud/c/README.md]: These set of lines should be deleted now, right?

File: clients/cloud/c/README.md [hunk 15]
```diff
@@ -2,17 +2,25 @@
...
+* Create a local file (e.g. at `librdkafka.config`) with configuration parameters to connect to your Kafka cluster, which can be on your local host, [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud), or any other cluster.  Follow [these detailed instructions](https://github.com/confluentinc/configuration-templates/tree/master/README.md) to properly create this file. 
 
-To run this example, create a local file with configuration parameters to connect to your Kafka cluster, which can be on your local host, [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud), or any other cluster.
 If this is a Confluent Cloud cluster, you must have:
 
 * Access to a [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud) cluster
```
Comment (ybyzek) [inline clients/cloud/c/README.md]: Since we are removing the lines below, let's combine this with above:

```
* If you are running on Confluent Cloud, you must have ccess to a [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud) cluster
```

File: clients/cloud/c/README.md (hunk 12 above)
Comment (laura-brouckman) [reply clients/cloud/c/README.md]: good catch, the rest of the time I use `$HOME/.confluent/librdkafka.config`, so I'll update this line as well

File: clients/cloud/kafka-commands/README.md [hunk 16]
```diff
@@ -11,10 +11,10 @@ To run this example, create a local file with configuration parameters to connec
 If this is a Confluent Cloud cluster, you must have:
 
 * Access to a [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud) cluster
-* Initialize a properties file at `$HOME/.ccloud/config` with configuration to your Confluent Cloud cluster:
+* Initialize a properties file at `$HOME/.confluent/config` with configuration to your Confluent Cloud cluster:
```
Comment (ybyzek) [inline clients/cloud/kafka-commands/README.md]: All READMEs with "Initialize a properties file..." needs to be replaced as was done for the clojure example.

File: clients/cloud/kafka-commands/README.md [hunk 17]
```diff
@@ -11,10 +11,10 @@ To run this example, create a local file with configuration parameters to connec
...
 
 ```shell
-$ cat $HOME/.ccloud/config
+$ cat $HOME/.confluent/config
```
Comment (ybyzek) [inline clients/cloud/kafka-commands/README.md]: All these files can be removed as well from READMEs.

File: clients/cloud/confluent-cli/README.md [hunk 18]
```diff
@@ -13,10 +13,10 @@ To run this example, create a local file with configuration parameters to connec
 If this is a Confluent Cloud cluster, you must have:
 
 * Access to a [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud) cluster
-* Initialize a properties file at `$HOME/.ccloud/config` with configuration to your Confluent Cloud cluster:
+* Initialize a properties file at `$HOME/.confluent/config` with configuration to your Confluent Cloud cluster:
```
Comment (ybyzek) [inline clients/cloud/confluent-cli/README.md]: This needs to be replaced as well, as was done for clojure.  And I don't think it will be called `$HOME/.confluent/config`, won't it be called `$HOME/.confluent/java.config`?

File: clients/cloud/confluent-cli/README.md [hunk 19]
```diff
@@ -34,12 +34,12 @@ The consumer reads the same topic from Confluent Cloud.
 1. Create the topic in Confluent Cloud
 
 ```bash
-$ kafka-topics --bootstrap-server `grep "^\s*bootstrap.server" $HOME/.ccloud/config | tail -1` --command-config $HOME/.ccloud/config --topic test1 --create --replication-factor 3 --partitions 6
+$ kafka-topics --bootstrap-server `grep "^\s*bootstrap.server" $HOME/.confluent/config | tail -1` --command-config $HOME/.confluent/config --topic test1 --create --replication-factor 3 --partitions 6
```
Comment (ybyzek) [inline clients/cloud/confluent-cli/README.md]: See comment above, need to verify file name

File: clients/cloud/java/src/main/java/io/confluent/examples/clients/cloud/ConsumerAvroExample.java [hunk 20]
```diff
@@ -46,10 +46,10 @@ public static void main(final String[] args) throws Exception {
     //   ssl.endpoint.identification.algorithm=https
     //   sasl.mechanism=PLAIN
     //   bootstrap.servers=<CLUSTER_BOOTSTRAP_SERVER>
-    //   sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="<CLUSTER_API_KEY>" password="<CLUSTER_API_SECRET>";
+    //   sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="{{ CLUSTER_API_KEY }}" password="{{ CLUSTER_API_SECRET }}";
```
Comment (ybyzek) [inline clients/cloud/java/src/main/java/io/confluent/examples/clients/cloud/ConsumerAvroExample.java]: Can we remove all this and replace it with a link to the new README?

e.g.

```
    // Load properties from a local configuration file
    // Create the configuration file (e.g. at '$HOME/.confluent/librdkafka.config') with configuration parameters
    // to connect to your Kafka cluster, which can be on your local host, Confluent Cloud, or any other cluster.
    // Follow these detailed instructions to properly create this file: https://github.com/confluentinc/configuration-templates/tree/master/README.md
```

Should be done for all instances, including:

java/src/main/java/io/confluent/examples/clients/cloud/StreamsExample.java:        // The configuration properties defined in the configuration file are assumed to include:
java/src/main/java/io/confluent/examples/clients/cloud/ConsumerAvroExample.java:    // The configuration properties defined in the configuration file are assumed to include:
java/src/main/java/io/confluent/examples/clients/cloud/ProducerAvroExample.java:    // The configuration properties defined in the configuration file are assumed to include:
java/src/main/java/io/confluent/examples/clients/cloud/StreamsAvroExample.java:        // The configuration properties defined in the configuration file are assumed to include:
java/src/main/java/io/confluent/examples/clients/cloud/ConsumerExample.java:    // The configuration properties defined in the configuration file are assumed to include:
java/src/main/java/io/confluent/examples/clients/cloud/ConsumerExamplePageviews.java:    // The configuration properties defined in the configuration file are assumed to include:
java/src/main/java/io/confluent/examples/clients/cloud/ProducerExample.java:    // The configuration properties defined in the configuration file are assumed to include:

File: clients/cloud/ksql-datagen/README.md [hunk 21]
```diff
@@ -14,10 +14,10 @@ To run this example, create a local file with configuration parameters to connec
 If this is a Confluent Cloud cluster, you must have:
 
 * Access to a [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud) cluster
-* Initialize a properties file at `$HOME/.ccloud/config` with configuration to your Confluent Cloud cluster:
+* Initialize a properties file at `$HOME/.confluent/config` with configuration to your Confluent Cloud cluster:
```
Comment (ybyzek) [inline clients/cloud/ksql-datagen/README.md]: Same -- needs to replaced & deleted (here and other places), like the clojure README

File: clients/cloud/nodejs/consumer.js [hunk 22]
```diff
@@ -28,8 +28,8 @@ function createConsumer(config, onData) {
     'bootstrap.servers': config['bootstrap.servers'],
     'sasl.username': config['sasl.username'],
     'sasl.password': config['sasl.password'],
-    'security.protocol': 'SASL_SSL',
-    'sasl.mechanisms': 'PLAIN',
+    'security.protocol': config['security.protocol'],
```
Comment (ybyzek) [inline clients/cloud/nodejs/consumer.js:31]: Where is it getting `security.protocol` and `sasl.mechanisms` from?  It's not currently in the config file.

Comment (laura-brouckman) [reply clients/cloud/nodejs/consumer.js:31]: ah i didn't notice those were removed from here https://github.com/confluentinc/configuration-templates/blob/master/clients/cloud/librdkafka.config. is there any reason to not have `sasl.mechanism` and `security.protocol` in the config files? We have them in the java-type ones, and it would help make the examples more consistent

Comment (ybyzek) [reply clients/cloud/nodejs/consumer.js:31]: @laura-brouckman  The examples repo used to have a mix (e.g. sometimes `security.protocol` was provided in the client config and sometimes in the code example itself) and because it was this unfortunate mix, that caused confusion.  

I see that you fixed that in your PR.

tl;dr my apologies, you have it the right way, and I will fix https://github.com/confluentinc/configuration-templates/blob/master/clients/cloud/librdkafka.config

Comment (ybyzek) [reply clients/cloud/nodejs/consumer.js:31]: Another great example of why implementing this single source of truth is so helpful.

Comment (ybyzek) [reply clients/cloud/nodejs/consumer.js:31]: @laura-brouckman , @rspurgeon  https://github.com/confluentinc/configuration-templates/pull/2

File: clients/cloud/ksql-datagen/README.md [hunk 23]
```diff
@@ -9,21 +9,8 @@ Produce messages to and consume messages from a Kafka cluster using the [KSQL da
...
-
-```shell
-$ cat $HOME/.ccloud/config
-bootstrap.servers=<BROKER ENDPOINT>
-ssl.endpoint.identification.algorithm=https
-security.protocol=SASL_SSL
-sasl.mechanism=PLAIN
-sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username\="<API KEY>" password\="<API SECRET>";
-```
+* Create a local file (e.g. at `$HOME/.confluent/java.config`) with configuration parameters to connect to your Kafka cluster, which can be on your local host, [Confluent Cloud](https://www.confluent.io/confluent-cloud/?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud), or any other cluster.  Follow [these detailed instructions](https://github.com/confluentinc/configuration-templates/tree/master/README.md) to properly create this file.
```
Comment (ybyzek) [inline clients/cloud/ksql-datagen/README.md:12]: ksql-datagen: 

* README references `$HOME/.confluent/java.config`
* But the script files reference `~/.confluent/java.config`

**Review by ybyzek — APPROVED** [review]
@laura-brouckman last final comment, after that LGTM

**Review by rspurgeon — APPROVED** [review]
LGTM thanks @laura-brouckman :boom:

**Comment by laura-brouckman** [comment]
@ybyzek could you merge this? I don't have write access to this repo
